dig.go

```diff
@@ -162,15 +198,57 @@ func setRand(r *rand.Rand) Option {
 	})
 }
 
+func (c *Container) knownTypes() []reflect.Type {
+	typeSet := make(map[reflect.Type]struct{}, len(c.providers))
+	for k := range c.providers {
+		typeSet[k.t] = struct{}{}
+	}
+
+	types := make([]reflect.Type, 0, len(typeSet))
+	for t := range typeSet {
+		types = append(types, t)
+	}
+	sort.Sort(byTypeName(types))
```

Comment:
Is this to make the output for suggestions deterministic?

Reply:
Yep. Deterministic output is easier to test.